src/gui/Application.cpp

```diff
@@ -262,3 +306,30 @@ bool Application::isAlreadyRunning() const
     return config()->get("SingleInstance").toBool() && m_alreadyRunning;
 }
 
+bool Application::sendFileNamesToRunningInstance(const QStringList& fileNames)
+{
+    QLocalSocket *client = new QLocalSocket(this);
+    connect(client, SIGNAL(disconnected()), client, SLOT(deleteLater()));
+
+    client->connectToServer(m_socketName);
+    const bool connected = client->waitForConnected(WaitTimeoutMSec);
+    if (connected) {
+        QByteArray data;
+        QDataStream out(&data, QIODevice::WriteOnly);
+        out.setVersion(QDataStream::Qt_5_0);
+        out << quint32(0)
+            << fileNames;
+        out.device()->seek(0);
+        out << quint32(data.size() - sizeof(quint32));
+
+        client->write(data);
+        client->flush();
+        client->waitForBytesWritten(WaitTimeoutMSec);
+        client->disconnectFromServer();
+    }  else {
+        client->deleteLater();
```

Comment:
You should move this to the top and add a return false after this statement.

Reply:
I did not understand what I should do :(

Reply:
Just move deleteLater() to the top, so you return early. Having a larger if block and then a simple else condition at the end of the function is quite error-prone and I consider it bad style. Better do whatever is in the else block right away and return and then you do what used to be in the if block.

Reply:
I rewrote the function